tests/cxx/badinc.cc

```diff
@@ -1841,7 +1841,6 @@ int main() {
   // IWYU: I1_Class needs a declaration
   H_TemplateFunction<I1_Class*>(&i1_class);
   H_TemplateFunction(&i1_class);
-  // IWYU: I1_Enum is...*badinc-i1.h
```

Comment:
Yes. At first it was a little bit surprising but I think it is correct to remove it. And we still need `I1_Enum` where we create `H_TemplateClass<I1_Enum> h_templateclass2(I11);`